phy/cluster/manual/wizard.py

```diff
@@ -67,62 +99,395 @@ def set_quality_function(self, func):
+
+    def next_best(self):
+        # Handle the case where we arrive at the end of the best list.
+        if self.best is not None and len(self._best_list) <= 1:
+            info("The wizard has finished.")
+        else:
+            self.best = _next(self._best_list,
+                              self._best,
+                              # self._is_not_ignored,
+                              )
+        if self.match is not None:
+            self._set_match_list()
+
+    def previous_best(self):
+        self.best = _previous(self._best_list,
+                              self._best,
+                              # self._is_not_ignored,
+                              )
+        if self.match is not None:
+            self._set_match_list()
+
+    def next_match(self):
+        # Handle the case where we arrive at the end of the match list.
+        if self.match is not None and len(self._match_list) <= 1:
+            self.next_best()
+        else:
+            self.match = _next(self._match_list,
+                               self._match,
+                               # self._is_not_ignored,
+                               )
+
+    def previous_match(self):
+        self.match = _previous(self._match_list,
+                               self._match,
+                               # self._is_not_ignored,
+                               )
+
+    def next(self):
+        if self.match is None:
+            return self.next_best()
+        else:
+            return self.next_match()
+
+    def previous(self):
+        if self.match is None:
+            return self.previous_best()
+        else:
+            return self.previous_match()
+
+    def first(self):
+        if self.match is None:
+            self.best = self._best_list[0]
+        else:
+            self.match = self._match_list[0]
+
+    def last(self):
+        if self.match is None:
+            self.best = self._best_list[-1]
+        else:
+            self.match = self._match_list[-1]
+
+    # Control
+    #--------------------------------------------------------------------------
+
+    def start(self):
+        self._set_best_list()
+
+    def pin(self, cluster=None):
+        if cluster is None:
+            cluster = self.best
+        if self.match is not None and self.best == cluster:
+            return
+        self.best = cluster
+        self._set_match_list(cluster)
+
+    def unpin(self):
+        if self.match is not None:
+            self.match = None
+            self._match_list = []
+
+    # Actions
+    #--------------------------------------------------------------------------
+
+    def _delete(self, clusters):
+        for clu in clusters:
+            if clu in self._cluster_groups:
+                del self._cluster_groups[clu]
+            if clu in self._best_list:
+                self._best_list.remove(clu)
+            if clu in self._match_list:
+                self._match_list.remove(clu)
+
+    def _add(self, clusters, group):
```

Comment:
not sure why these should be private functions. Sure, they're rare operations, but still possible for the user to do, right?